examples/version-migration/schema-upgrade/src/modelVersion1/containerCode.ts

```diff
@@ -3,80 +3,82 @@
  * Licensed under the MIT License.
  */
 
-import type { IMigrationTool } from "@fluid-example/example-utils";
 import {
-	MigrationToolFactory,
-	ModelContainerRuntimeFactory,
 	getDataStoreEntryPoint,
+	instantiateMigratableRuntime,
 } from "@fluid-example/example-utils";
-import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type {
+	IContainer,
+	IContainerContext,
+	IRuntime,
+	IRuntimeFactory,
+} from "@fluidframework/container-definitions/internal";
+import type { IContainerRuntimeOptions } from "@fluidframework/container-runtime/internal";
 import type { IContainerRuntime } from "@fluidframework/container-runtime-definitions/internal";
 
-import type { IInventoryList, IInventoryListAppModel } from "../modelInterfaces.js";
+import type { IInventoryList } from "../modelInterfaces.js";
 
 import { InventoryListAppModel } from "./appModel.js";
 import { InventoryListInstantiationFactory } from "./inventoryList.js";
 
 const inventoryListId = "default-inventory-list";
-const migrationToolId = "migration-tool";
-
-const migrationToolRegistryKey = "migration-tool";
-const migrationToolFactory = new MigrationToolFactory();
 
 /**
  * @internal
  */
-export class InventoryListContainerRuntimeFactory extends ModelContainerRuntimeFactory<IInventoryListAppModel> {
+export class InventoryListContainerRuntimeFactory implements IRuntimeFactory {
+	public get IRuntimeFactory() {
```

Comment:
Now that the factory isn't a ModelContainerRuntimeFactory, there's a little more boilerplate, but the customer gains control of lifecycle and inheritance as they see fit.